src/server/system_services/system_server.js

```diff
@@ -1282,6 +1282,24 @@ async function _get_endpoint_groups() {
     });
 }
 
+/**
+ *
+ * ROTATE ROOT KEY
+ *
+ */
+async function rotate_root_key(req) {
+    const new_secret = req.rpc_params.new_secret;
+    const mkm = system_store.master_key_manager;
```

Comment:
Why add a new function instead of using rotate_master_key and adding a new entity?

Reply:
as discussed... not the same flow and don't want to just add extra code to an already huge function